core/tests/protractor_desktop/learnerDashboard.js

```diff
@@ -153,19 +208,234 @@ describe('Learner dashboard functionality', function() {
+
+    await topicEditorPage.expectNumberOfStoriesToBe(1);
+    await topicEditorPage.expectStoryPublicationStatusToBe('No', 0);
+    await topicEditorPage.navigateToStoryWithIndex(0);
+    await storyEditorPage.updateMetaTagContent('story meta tag');
+    await storyEditorPage.saveStory('Added thumbnail.');
+    await storyEditorPage.publishStory();
+    await storyEditorPage.returnToTopic();
+    await topicEditorPage.expectStoryPublicationStatusToBe('Yes', 0);
+
+    var url = await browser.getCurrentUrl();
+    var topicId = url.split('/')[4].slice(0, -1);
+    await general.closeCurrentTabAndSwitchTo(handle);
+    await adminPage.editConfigProperty(
+      'The details for each classroom page.',
+      'List',
+      async function(elem) {
+        elem = await elem.editItem(0, 'Dictionary');
+        elem = await elem.editEntry(4, 'List');
+        elem = await elem.addItem('Unicode');
+        await elem.setValue(topicId);
+      });
+    await topicsAndSkillsDashboardPage.get();
+    (
+      await
+      topicsAndSkillsDashboardPage.createSkillWithDescriptionAndExplanation(
+        'Skill 1', 'Concept card explanation', false));
+    await topicsAndSkillsDashboardPage.get();
+    await topicsAndSkillsDashboardPage.navigateToSkillsTab();
+    await topicsAndSkillsDashboardPage.assignSkillToTopic(
+      'Skill 1', 'Topic 1');
+    await topicsAndSkillsDashboardPage.get();
+    await topicsAndSkillsDashboardPage.navigateToTopicWithIndex(0);
+    await topicEditorPage.addSubtopic(
+      'Subtopic 1', 'subtopic-one', '../data/test2_svg.svg',
+      'Subtopic content');
+    await topicEditorPage.saveTopic('Added subtopic.');
+
+    await topicEditorPage.navigateToTopicEditorTab();
+    await topicEditorPage.navigateToReassignModal();
+
+    await topicEditorPage.dragSkillToSubtopic('Skill 1', 0);
+    await topicEditorPage.saveRearrangedSkills();
+    await topicEditorPage.saveTopic('Added skill to subtopic.');
+
+    await topicEditorPage.updateMetaTagContent('meta tag content');
+    await topicEditorPage.updatePageTitleFragment('fragment');
+    await topicEditorPage.saveTopic('Added meta tag.');
+
+    await topicEditorPage.publishTopic();
+    await learnerDashboardPage.get();
+    await learnerDashboardPage.navigateToHomeSection();
+    await learnerDashboardPage.expectCountOfTopicInSuggestedForYou(1);
+    await learnerDashboardPage.navigateToGoalsSection();
+    await driver.findElement(
+      by.css('.protractor-test-edit-goals-section'));
+    await learnerDashboardPage.expectNameOfTopicInEditGoalsToMatch(
+      TOPIC_NAME);
+    await driver.findElement(By.css(
+      '.protractor-test-add-topic-to-current-goals-button')).click();
+    await learnerDashboardPage.navigateToGoalsSection();
+    await learnerDashboardPage.expectNameOfTopicInCurrentGoalsToMatch(
+      `Learn ${TOPIC_NAME}`);
+    await learnerDashboardPage.navigateToHomeSection();
+    await learnerDashboardPage.expectCountOfTopicInContinueWhereYouLeftOff(1);
```

Comment:
Check 'Suggested for you' section as well? (It should be 0, right?)

Reply:
We can't check this here. It's because what `expectCountOfTopicInContinueWhereYouLeftOff` does is it waits for the visibility of LearnerTopicSummaryTile in the suggested for you section and counts the number of tiles in it. Since the number of topics in the `Suggested for you` section will be 0, no LearnerTopicSummaryTile will be displayed and hence it will give an error something like 'Learner Topic Name takes too long to appear'.